specification/schemaregistry/SchemaRegistry/main.tsp

```diff
@@ -0,0 +1,267 @@
+@server(
+  "{fullyQualifiedNamespace}",
+  "The Schema Registry service endpoint.",
+  {
+    @doc("The Schema Registry service endpoint, for example 'https://my-namespace.servicebus.windows.net'.")
+    fullyQualifiedNamespace: url,
+  }
+)
+@versioned(ServiceApiVersions)
+@doc("SchemaRegistryClient is a client for registering and retrieving schemas from the Azure Schema Registry service.")
+namespace SchemaRegistry;
+
+@doc("Represents the Schema Registry API version to use for requests.")
+enum ServiceApiVersions {
+  @useDependency(Azure.Core.Versions.v1_0_Preview_2)
+  @doc("Azure Schema Registry 2021-10 Version")
+  V2021_10: "2021-10",
+
+  @useDependency(Azure.Core.Versions.v1_0_Preview_2)
+  @doc("Azure Schema Registry 2022-10 Version")
+  V2022_10: "2022-10",
+
+  @useDependency(Azure.Core.Versions.v1_0_Preview_2)
+  @doc("Azure Schema Registry 2023-07-01 Version. This is the default version.")
+  V2023_07_01: "2023-07-01",
+}
+
+// ROUTES
+alias resourceOperations = Azure.Core.StandardResourceOperations;
+@client({
+  name: "SchemaRegistryClient",
+  service: SchemaRegistry,
+})
+interface SchemaOperations {
+  //  GET https://{namespaceName}.servicebus.windows.net/$schemaGroups?api-version={apiVersion}
+  #suppress "@azure-tools/typespec-azure-core/use-standard-operations" "Standard read operation w/ SchemaId resource doesn't allow returning binary schema and headers."
+  @summary("Get list of schema groups.")
+  @doc("Gets the list of schema groups user is authorized to access.")
+  @route("/$schemaGroups")
+  listSchemaGroups is Azure.Core.Foundations.Operation<{}, StringsList, {}>;
+
+  //  GET https://{namespaceName}.servicebus.windows.net/$schemaGroups/$schemas/{id}?api-version={apiVersion}
+  #suppress "@azure-tools/typespec-azure-core/use-standard-operations" "Standard read operation w/ SchemaId resource doesn't allow returning binary schema and headers."
+  @summary("Get a registered schema by its unique ID reference.")
+  @doc("Gets a registered schema by its unique ID.  Azure Schema Registry guarantees that ID is unique within a namespace. Operation response type is based on serialization of schema requested.")
+  @route("/$schemaGroups/$schemas/{id}")
+  @get
+  getSchemaById is Azure.Core.Foundations.Operation<
+    Azure.Core.Foundations.ItemKeysOf<SchemaId>,
+    OkResponse & SchemaHeaders & SchemaContent & SchemaContentType
+  >;
+
+  //  GET https://{namespaceName}.servicebus.windows.net/$schemaGroups/{groupName}/schemas/{name}/versions?api-version={apiVersion}
+  #suppress "@azure-tools/typespec-azure-core/use-standard-operations" "Standard read operation w/ Version resource doesn't allow returning non-version response."
+  @summary("List schema versions.")
+  @doc("Gets the list of all versions of one schema.")
+  @route("/$schemaGroups/{groupName}/schemas/{name}/versions")
+  listSchemaVersions is Azure.Core.Foundations.Operation<
```

Comment:
Use PagedOperation here instead of regular Operation

Reply:
using the standardOperation ListResource should work here in the future. At this point, this is the best we can do.

Reply:
open a bug in typespec autorest about this. expected.

Reply:
issue here: https://github.com/Azure/typespec-azure/issues/229